_adr/86/index.adoc

```diff
@@ -13,21 +13,87 @@ tags:
+
+```
+   - authentication:
+      customClaimCheck: '@.rh-org-id == ''testorgid''|| @.org_id == ''testorgid'''
+      tokenEndpointUri: >-
+            https://tokenendpointauri.com/token
+       ...
+    serviceAccounts:
+    - name: canary
+      password: **********
+      principal: canary-client-Id    
+```
+
+
+Strimzi Canary uses SASL/Plain authentication mechanism with the Kafka broker. Control plane i.e. KFM provides a canary service account client ID and secret in the Managed Kafka Custom Resource (CR) along with a Token Endpoint URI (Uniform Resource Identifier). Canary uses this information to authenticate with the Kafka brokers using SASL/Plain authentication mechanism, i.e. similar to username and password. Canary provides the client ID and secret as a part of its authentication request to the Kafka Broker. Kafka broker uses this information to authenticate with the service mentioned in the token endpoint URI. Currently, this token end point URI is configured to be the MAS SSO service. 
+
+Upon a successful authentication, MAS SSO provides a valid token back to the Kafka broker. Kafka brokers as a part of its authentication flow, upon a successful response, it introspects the token received and performs a custom claims check. It verifies that the "Canary Service account Client" belongs to the same organization as the creator or owner of the Kafka instance. Kafka brokers deem the authentication to be successful after this.
+
+image::adr-86-current-authentication-flow.png[Current authenticaion flow] 
+
 ## Proposed architecture
 
+When using sso.redhat.com as the single authentication post migration, KFM will create the Canary Service Account Clients in a "Special Management Organization". This would mean all the internal service accounts such as the Canary for all users (for all organizations) are created in this single organization that is managed and owned by Red hat. The reasons for this are, the limit on the number of service accounts an organization can create do not apply for this organization, and it provides more security by restricting these internal service accounts to the organization that is owned by red hat rather than the customer's organization.
+
+This would mean that all the Canary service accounts for all customers/users will be created/associated with this "Special Management Organization" when compared to the existing architecture where they were scoped to the user's organization. This requires some additional steps in authentical flow described in the current architecture section.
+
+- KFM when building the "CustomClaimCheck" includes the  Canary service account client ID "clientID" field as well
+
+```
+   - authentication:
+      customClaimCheck: '@.rh-org-id == ''testorgid''|| @.org_id == ''testorgid'' || @.clientId == ''canary-client-Id'''
```

Comment:
it will fail for external clients/users as they will only satisfy the org id check. Canary accounts is going to have different org id.

Reply:
That is correct.. All internal service accounts will not be a part of Customers Org ID. Resolving this.

Reply:
So I might have got me example wrong, but surely the `@.clientId == ''canary-client-Id'''` should apply only when the org id _is also_ this special RH internal org for the service account? What about an account with `clientId == canary-client-id`, but in a different org?

Reply:
Since all Canaries will be created in the same Org (Special management Org) that seemed a bit redundant in our use case. All internal service accounts are created in "Special management ORg"